code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -1165,20 +1223,18 @@
 
 /datum/reagent/medicine/psicodine/on_mob_metabolize(mob/living/L)
 	..()
-	ADD_TRAIT(L, TRAIT_FEARLESS, type)
+	ADD_TRAIT(L, TRAIT_FEARLESS, PSYCH_TRAIT)
 
 /datum/reagent/medicine/psicodine/on_mob_end_metabolize(mob/living/L)
-	REMOVE_TRAIT(L, TRAIT_FEARLESS, type)
+	REMOVE_TRAIT(L, TRAIT_FEARLESS, PSYCH_TRAIT)
 	..()
 
 /datum/reagent/medicine/psicodine/on_mob_life(mob/living/carbon/M)
 	M.jitteriness = max(0, M.jitteriness-6)
 	M.dizziness = max(0, M.dizziness-6)
 	M.confused = max(0, M.confused-6)
 	M.disgust = max(0, M.disgust-6)
-	var/datum/component/mood/mood = M.GetComponent(/datum/component/mood)
-	if(mood.sanity <= SANITY_NEUTRAL) // only take effect if in negative sanity and then...
-		mood.setSanity(min(mood.sanity+5, SANITY_NEUTRAL)) // set minimum to prevent unwanted spiking over neutral
+	SEND_SIGNAL(M,COMSIG_SET_SANITY,SANITY_NEUTRAL)
```

Comment:
is this changing functionality, I can't tell

Reply:
no. it is just code improvement. You shouldn't use GetComponent

Reply:
I see now, I'm a big dummy